apptools/persistence/state_pickler.py

```diff
@@ -341,8 +360,19 @@ def _do(self, obj):
             return self._do_dict(obj)
         elif hasattr(obj, '__dict__'):
             return self._do_instance(obj)
+        else:
+            warnings.warn("Cannot pickle unrecognized type {}. Returning None"
+                          " for backward compatibility.".format(obj_type))
+            return None
 
     def _get_id(self, value):
+        # We consider a special case for numpy scalar values, because
+        # they hash as native types, but they are special because we
+        # want to recover their true type, and the only way of doing so
+        # is to consider them as objects.
+        if isinstance(value, numpy.generic):
+            return id(value)
```

Comment:
This is a small problem with using `id` here. The same `id` can be re-used when the original object is garbaged collected. So if during the pickling process the numpy arrays are garbage collected then the registry might point to the wrong object.